structured_external_storage_all/models/ir_attachment.py

```diff
@@ -23,8 +23,14 @@ def create_metadata(self):
             # check if res_model is in the define rules.
             if attachment.res_model in all_rules:
                 sync_rule = sync_rule_obj.search(
-                    [('source_model.object', '=', attachment.res_model)],
-                    order='sequence asc', limit=1)
+                    [('source_model.object', '=', attachment.res_model),
+                     ('sequence', '<', 10)],
+                    order='sequence asc')
+                if not sync_rule:
+                    sync_rule = sync_rule_obj.search(
+                        [('source_model.object', '=', attachment.res_model)],
+                        order='sequence asc', limit=1)
+
```

Comment:
This is according to the first idea that we discussed. But, after discussing with GJ, we thought it would be more intuitive to implement the idea a tiny bit differently, as I said on Telegram also.

1. Rules with the same sequence are always executed no matter what
2. Rules with a higher sequence (=lower priority) are only executed when there has been no matching rule with the same location.

This is easily implemented by using the following algorithm:

```
all_matching_rules = sync_rule_obj.search([
    ('source_model.object', '=', attachment.res_model)
], order='sequence asc')
selected_matching_rules = []
for rule in all_matching_rules:
    previous_rules_with_same_loc_and_lower_seq = [
        previous_rule for previous_rule in selected_matching_rules
        if previous_rule.location_id == rule.location_id
        and previous_rule.sequence < rule.sequence
    ]
    if not previous_rules_with_same_loc_and_lower_seq:
        selected_matching_rules.append(rule)
for rule in selected_matching_rules:
    #.... create meta for this rule
```